### ISSUE TITLE
deleteByExample error using table alias, ANNOTATEDMAPPER and mysql 

### ISSUE THREAD

**Issue by italobb** [mybatis/generator#57]
If the table has an `alias` associated, mybatis-generator is using that `alias` in the `deleteByExample`, what is not accepted by MySQL in `delete` clauses. 

Example. Consider a mybatis-generator config like this:

```
<generatorConfiguration>
   ...
   <context ...>
      ...
      <javaClientGenerator type="ANNOTATEDMAPPER" .../>
      <table tableName="myTable" alias="mt"/>
      ...
   </context>
</generatorConfiguration>
```

Generate the mapper and run the method bellow using MySQL:

```
myTableMapper.deleteByExample(example);
```

The following error will be shown:

```
### Error updating database.  Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'mt
WHERE ((mt.field = 'value'))' at line 1
### The error may involve com.test.mapper.MyTableMapper.deleteByExample-Inline
### The error occurred while setting parameters
### SQL: DELETE FROM myTable mt WHERE ((mt.field = ?))
### Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'mt
WHERE ((mt.field = 'value'))' at line 1
```

Because the generated SQL do not complies with [MySQL DELETE syntax](http://dev.mysql.com/doc/refman/5.7/en/delete.html):

```
DELETE FROM myTable mt WHERE ((mt.field = ?))
```

I think the corret generated `DELETE` clause should be like:

```
DELETE mt FROM myTable mt WHERE ((mt.field = ?)) 
```

Is that a known bug? How to fix that? I need to keep the alias use in the other queries. [This question is at StackOverflow too](http://stackoverflow.com/questions/33435970/deletebyexample-error-using-table-alias-mysql-and-annotatedmapper).

**Comment by jeffgbutler** [comment]
It is a known issue.  It only arises on certain databases like MySQL.  I think the best solution would be to change the generate so the alias is only used on select statements and not anywhere else.  I'll take a look at making that fix.

**Comment by italobb** [comment]
The behaviour with MAPPER option, instead of ANNOTATEDMAPPER, was not to use the alias in DELETE clause. So, I agree with you, it can be just ignored and will be compatible with old codes that used MAPPER.

**Comment by LouisWayne** [comment]
Still happening - any updates?

**Comment by jeffgbutler** [comment]
This turns out to be difficult to solve with the generator as it stands now.  I've been working on a new SQL templating library for MyBatis and I have a fix for this issue in that new library.  I think that's how it's ultimately going to be fixed because it would just be too cumbersome to fix it in the current generator.

**Comment by jeffgbutler** [comment]
This will not be resolved in what I'm now calling the "legacy" code generators - it would mean generating a huge amount of new code.

This IS resolved in the new code generator that is based on MyBatis Dynamic SQL that was just released.

**Comment by loveychao** [comment]
Which version can solve this problem?

**Comment by jeffgbutler** [comment]
It is fixed in the latest version (1.3.6) - but only if you change to the new Dynamic SQL runtime.  You do that by specifying `<context targetRuntime"MyBatis3DynamicSql">`

If you do this, please note that the generated code is not 100% compatible with prior versions.  I think it is much better and far more flexible. If you have not made the change to this new runtime yet, then I recommend you read about it first to see if this is a good option for you. http://www.mybatis.org/generator/generatedobjects/dynamicSql.html
